test/e2e/mock-response-data/snaps/update-snap-binary.ts

```diff
@@ -0,0 +1,215 @@
+    headersFilePath,
+    `${JSON.stringify(relevantHeaders, null, 2)}\n`,
+  );
+  console.log(`${GREEN}Successfully saved:${RESET} ${txtFilePath}`);
+  console.log(`${GREEN}Successfully saved:${RESET} ${headersFilePath}`);
+};
+
+const deleteOldSnapFiles = (
+  snapName: string,
+  currentVersionBeingSaved: string,
+) => {
+  let versionToDelete: string | null = null;
+
+  try {
+    const latestExistingVersion = getLocalSnapLatestVersion(snapName);
+
+    if (latestExistingVersion !== currentVersionBeingSaved) {
+      versionToDelete = latestExistingVersion;
+    }
+  } catch (error: unknown) {
+    console.log(
+      `${YELLOW}Info: No existing versions of ${snapName} found to check for deletion`,
+    );
+  }
+
+  if (versionToDelete) {
+    console.log(
+      `${CYAN}Identified version to delete for ${snapName}: ${versionToDelete}${RESET}`,
+    );
+    const txtFilePath = path.join(
+      SNAP_DIR,
+      `${snapName}-${versionToDelete}.txt`,
+    );
+    const headersFilePath = path.join(
+      SNAP_DIR,
+      `${snapName}-${versionToDelete}-headers.json`,
+    );
+
+    [txtFilePath, headersFilePath].forEach((filePathToDelete) => {
+      if (fs.existsSync(filePathToDelete)) {
+        try {
+          fs.unlinkSync(filePathToDelete);
+          console.log(
+            `${GREEN}Successfully deleted old file:${RESET} ${filePathToDelete}`,
+          );
+        } catch (e) {
+          console.error(
+            `${RED}Error deleting file ${filePathToDelete}:${RESET}`,
+            e,
+          );
+        }
+      }
+    });
+  }
+};
+
+const printHelp = () => {
+  console.log(
+    `${YELLOW}Usage:${RESET} yarn update-snap-binary ${CYAN}--<snapName>-<version>${RESET}`,
+  );
+  console.log(
+    `${YELLOW}Example:${RESET} yarn update-snap-binary ${CYAN}--bip32-example-snap-2.3.0${RESET}`,
+  );
+  console.log(
+    `Please ensure the version format is x.y.z (e.g., 1.2.3)${RESET}`,
+  );
+};
+
+const main = async () => {
+  const args = getArgs();
+
+  if (args.help) {
+    printHelp();
+    process.exit(0);
+  }
+
+  if (!args.snapNameAndVersion) {
+    console.error(
+      `${RED}Error: Snap name and version not provided. Use --help for usage.${RESET}`,
+    );
+    process.exit(1);
+  }
+
+  const nameAndVersionArg = args.snapNameAndVersion;
+  const parts = nameAndVersionArg.split('-');
+
+  if (parts.length < 2) {
+    console.error(
+      `${RED}Error: Invalid format for snap name and version: "${nameAndVersionArg}". Expected format: <snapName>-<version> (e.g., bip32-example-snap-2.3.0). Use --help for usage.${RESET}`,
+    );
+    process.exit(1);
+  }
+
+  const versionCandidate = parts[parts.length - 1];
+  const snapNameParts = parts.slice(0, -1);
+
+  if (!/^\d+\.\d+\.\d+$/u.test(versionCandidate)) {
+    console.error(
+      `${RED}Error: Invalid version format in "${nameAndVersionArg}". Version part "${versionCandidate}" is not in x.y.z format. Use --help for usage.${RESET}`,
+    );
+    process.exit(1);
+  }
```

Comment:
The `semver` package can help here as well.

Reply:
updated ! now using `semverValid